docs/proposals/001-proposal-template.md

```diff
@@ -0,0 +1,92 @@
+---
+title: Neat-enhancement-idea
+authors:
+  - "@sbose78"
+reviewers:
+  - "@alexmt"
+  - TBD
+approvers:
+  - "@alexmt"
+  - TBD
+
+creation-date: yyyy-mm-dd
+last-updated: yyyy-mm-dd
+---
+
+# Neat Enhancement Idea
+
+This is the title of the enhancement. Keep it simple and descriptive. A good title can help
+communicate what the enhancement is and should be considered as part of any review.
+
+
+## Open Questions [optional]
+
+This is where to call out areas of the design that require closure before deciding to implement the
+design.
+
+
+## Summary
+
+The `Summary` is required for producing accurate user-focused documentation
+such as release notes or a development roadmap. It should be possible to collect this information
+before implementation begins in order to avoid requiring implementors to split their attention
+between writing release notes and implementing the feature itself.
+
+
+## Motivation
+
+This section is for explicitly listing the motivation, goals and non-goals of this proposal.
+Describe why the change is important and the benefits to users.
+
```

Comment:
It might be worth adding a specific "Use Case (Details)" (sub)section to make sure we catch details on what the proposal is trying to solve, without getting too deep into technical/implementation details. This will provide more and valuable background to the goals.